### ISSUE TITLE
Swift emits an invalid module interface when a public type has the same name as a module

### ISSUE THREAD

**Issue by Alexandeer** [marmelroy/PhoneNumberKit#755]
When building a library that depends on PhoneNumberKit, I receive warnings:

`warning: public class 'PhoneNumberKit.PhoneNumberKit' shadows module 'PhoneNumberKit', which may cause failures when importing 'PhoneNumberKit' or its clients in some configurations; please rename either the class 'PhoneNumberKit.PhoneNumberKit' or the module 'PhoneNumberKit', or see https://github.com/apple/swift/issues/56573 for workarounds`

[I think no further clarification is required](https://github.com/apple/swift/issues/56573). I also want to notify that I observed a problem when the project did not understand definition like PhoneNumberKit.CountryCodePicker. There was an error with text like the PhoneNumberKit module does not contain CountryCodePicker (it is part of the PhoneNumberKit class). 
In short, Xcode confuses where the class is and where the module is.

**Comment by 1ec5** [comment]
Darn, it’s still an issue…
